Assets/BossRoom/Scripts/Shared/Net/UnityServices/Lobbies/LocalLobby.cs

```diff
@@ -0,0 +1,290 @@
+            }
+
+            DoAddPlayer(user);
+            OnChanged(this);
+        }
+
+        private void DoAddPlayer(LobbyUser user)
+        {
+            m_LobbyUsers.Add(user.ID, user);
+            user.onChanged += OnChangedUser;
+        }
+
+        public void RemovePlayer(LobbyUser user)
+        {
+            DoRemoveUser(user);
+            OnChanged(this);
+        }
+
+        private void DoRemoveUser(LobbyUser user)
+        {
+            if (!m_LobbyUsers.ContainsKey(user.ID))
+            {
+                Debug.LogWarning($"Player {user.DisplayName}({user.ID}) does not exist in lobby: {LobbyID}");
+                return;
+            }
+
+            m_LobbyUsers.Remove(user.ID);
+            user.onChanged -= OnChangedUser;
+        }
+
+        private void OnChangedUser(LobbyUser user)
+        {
+            OnChanged(this);
+        }
+
+        public string LobbyID
+        {
+            get => m_data.LobbyID;
+            set
+            {
+                m_data.LobbyID = value;
+                OnChanged(this);
+            }
+        }
+
+        public string LobbyCode
+        {
+            get => m_data.LobbyCode;
+            set
+            {
+                m_data.LobbyCode = value;
+                OnChanged(this);
+            }
+        }
+
+        public string RelayCode
+        {
+            get => m_data.RelayCode;
+            set
+            {
+                m_data.RelayCode = value;
+                OnChanged(this);
+            }
+        }
+
+        public string RelayNGOCode
+        {
+            get => m_data.RelayNGOCode;
+            set
+            {
+                m_data.RelayNGOCode = value;
+                m_data.RelayNGOCode_LastEdit = DateTime.Now.Ticks;
+                OnChanged(this);
+            }
+        }
+
+        public string LobbyName
+        {
+            get => m_data.LobbyName;
+            set
+            {
+                m_data.LobbyName = value;
+                OnChanged(this);
+            }
+        }
+
+        public LobbyState State
+        {
+            get => m_data.State;
+            set
+            {
+                m_data.State = value;
+                m_data.State_LastEdit = DateTime.Now.Ticks;
+                OnChanged(this);
+            }
+        }
+
+        public bool Private
+        {
+            get => m_data.Private;
+            set
+            {
+                m_data.Private = value;
+                OnChanged(this);
+            }
+        }
+
+        public int PlayerCount => m_LobbyUsers.Count;
+
+        public int MaxPlayerCount
+        {
+            get => m_data.MaxPlayerCount;
+            set
+            {
+                m_data.MaxPlayerCount = value;
+                OnChanged(this);
+            }
+        }
+
+        public OnlineMode OnlineMode
+        {
+            get => m_data.OnlineMode;
+            set
+            {
+                if (m_data.OnlineMode != value)

```

Comment:
This is the only one that you check if the value has changed before triggering `OnChanged`.  Should the other properties have it as well?